include/ftxui/component/component_options.hpp

```diff
@@ -177,6 +177,7 @@ struct InputOption {
   std::function<Element(InputState)> transform;
   Ref<bool> password = false;  /// < Obscure the input content using '*'.
   Ref<bool> multiline = true;  /// < Whether the input can be multiline.
+  Ref<bool> overwrite = false; /// < Overwrite text at cursor.
```

Comment:
I think developers are more aware of the "insert" mode than the "overtype" one.

What about using:
```cpp
  Ref<bool> insert = true;     /// < Insert or overtype character mode.
```